README.md

```diff
@@ -9,185 +10,157 @@
+| ✅      | [Named clients](#named-clients)                 | Utilize multiple providers in a single application.                                 |
+| ✅      | [Eventing](#eventing)                              | Listen for state changes in the provider or flag management system and react on the events.                |
+| ✅      | [Shutdown](#shutdown)                | Gracefully clean up a provider during application shutdown.             |
+| ✅      | [Extending](#extending)       | Extend OpenFeature with custom providers and hooks to support your exact use case                        |
+
+<sub>Implemented: ✅ | Partially implemented: ⚠️ | Not implemented yet: ❌</sub>
+
+### Providers
+
+[Providers](https://openfeature.dev/docs/reference/concepts/provider) are an abstraction between a flag management system and the OpenFeature SDK.
+Look [here](https://openfeature.dev/ecosystem?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Provider&instant_search%5BrefinementList%5D%5Btechnology%5D%5B0%5D=Go) for a complete list of available providers.
+If the provider you're looking for hasn't been created yet, see the [develop a provider](#develop-a-provider) section to learn how to build it yourself.
+
+Once you've added a provider as a dependency, it can be registered with OpenFeature like this:
+
+```go
+openfeature.SetProvider(MyProvider{})
+```
 
-For complete documentation, visit: https://openfeature.dev/docs/category/concepts
+In some situations, it may be beneficial to register multiple providers in the same application.
+This is possible using [named clients](#named-clients), which is covered in more details below.
 
-### Context-aware evaluation:
+### Context-aware evaluation
 
-Sometimes the value of a flag must take into account some dynamic criteria about the application or user, such as the user location, IP, email address, or the location of the server.
+Sometimes, the value of a flag must consider some dynamic criteria about the application or user, such as the user's location, IP, email address, or the server's location.
 In OpenFeature, we refer to this as [`targeting`](https://openfeature.dev/specification/glossary#targeting).
-If the flag system you're using supports targeting, you can provide the input data using the `EvaluationContext`.
+If the flag management system you're using supports targeting, you can provide the input data using the `EvaluationContext`.
 
 ```go
-// add a value to the global context
+// set a value to the global context
 openfeature.SetEvaluationContext(openfeature.NewEvaluationContext(
-    "foo",
+    "",
     map[string]interface{}{
-        "myGlobalKey":  "myGlobalValue",
+        "region":  "us-east-1-iah-1a",
     },
 ))
 
-// add a value to the client context
+// set a value to the client context
 client := openfeature.NewClient("my-app")
 client.SetEvaluationContext(openfeature.NewEvaluationContext(
     "", 
```

Comment:
Not related to this readme but I found it weird to send an empty string here.
We may need to add a comments in the example to explain why we send this empty string.

Reply:
Is this the targeting key? If so I think we should "make one up". I think that will be clearer.

Reply:
Yes, I agree this is probably the best we can do here.

I was looking at the [`EvaluationContext` struct](https://github.com/open-feature/go-sdk/blob/main/pkg/openfeature/evaluation_context.go)  and I think we can add a new constructor, something like this:

```go
func NewEvaluationContextWithoutTargetingKey(attributes map[string]interface{}) EvaluationContext {
	// copy attributes to new map to avoid reference being externally available, thereby enforcing immutability
	attrs := make(map[string]interface{}, len(attributes))
	for key, value := range attributes {
		attrs[key] = value
	}

	return EvaluationContext{
		targetingKey: "",
		attributes:   attrs,
	}
}
```

This will allow us to create a context without requiring a `targetingKey`.
_The naming of the function is bad, but this is the 1st thing coming to my mind_.

Reply:
We could do `KeylessContext` `TargetlessContext` `BaseContext`

Reply:
I have open a PR for this https://github.com/open-feature/go-sdk/pull/204

Reply:
I've updated the readme to reflect these changes. Thanks for the help @thomaspoignant 🥳